app/src/main/java/com/celzero/bravedns/service/ConnectionMonitor.kt

```diff
@@ -0,0 +1,240 @@
+    }
+
+    override fun onCapabilitiesChanged(network: Network, networkCapabilities: NetworkCapabilities) {
+        if(DEBUG) Log.d(LOG_TAG, "ConnectionMonitor - onCapabilitiesChanged")
+        handleNetworkChange(isForceUpdate = false)
+    }
+
+    override fun onLinkPropertiesChanged(network: Network, linkProperties: LinkProperties) {
+        if(DEBUG) Log.d(LOG_TAG, "ConnectionMonitor - onLinkPropertiesChanged")
+        handleNetworkChange(isForceUpdate = false)
+    }
+
+    /**
+     * Called when the preference(isAddAllNetwork) is changed.
+     * if the preference is true, add only active network
+     * else, add all available network.
+     */
+    fun onUserPreferenceChanged(){
+        if (DEBUG) Log.d(LOG_TAG, "ConnectionMonitor - onUserPreferenceChanged")
+        handleNetworkChange(isForceUpdate = true)
+    }
+
+    /**
+     * Returns the network list based on the user preference.
+     * If isAddAllNetwork preference is selected then returns the list of available networks.
+     * else return the current active network.
+     */
+    fun getNetworkList(): LinkedHashSet<Network> {
+        val networks: LinkedHashSet<Network> = linkedSetOf()
+        if (persistentState.isAddAllNetworks) {
+            networks.addAll(currentNetworks)
+        } else {
+            currentActiveNetwork?.let { networks.add(it) }
+        }
+        return networks
+    }
+
+    private fun handleNetworkChange(isForceUpdate : Boolean) {
+        val message = constructMessage(if (persistentState.isAddAllNetworks) MSG_ADD_ALL_NETWORKS else MSG_ADD_ACTIVE_NETWORK, isForceUpdate)
+        serviceHandler?.removeMessages(MSG_ADD_ACTIVE_NETWORK, null)
+        serviceHandler?.removeMessages(MSG_ADD_ALL_NETWORKS, null)
+        serviceHandler?.sendMessage(message)
+    }
+
+    /**
+     * Constructs the message object for Network handler.
+     * Add the active network to the message object
+     * in case of setUnderlying network has only active networks.
+     */
+    private fun constructMessage(what: Int, isForceUpdate: Boolean): Message {
+        val message = Message.obtain()
+        message?.what = what
+        message?.obj = isForceUpdate
+        return message!!
+    }
+
+    // Handles the network messages from the callback from the connectivity manager
+    private class NetworkRequestHandler(looper: Looper?, val connectionMonitor: ConnectionMonitor) : Handler(looper!!) {
+        override fun handleMessage(msg: Message) {
+            // isForceUpdate - true if onUserPreferenceChanged is changes, the messages should be
+            // processed forcefully regardless of the current and new networks.
+            val isForceUpdate : Boolean = msg.obj as Boolean
+            when(msg.what) {
+                MSG_ADD_ACTIVE_NETWORK -> processActiveNetwork(isForceUpdate)
+                MSG_ADD_ALL_NETWORKS -> processAllNetworks(isForceUpdate)
+            }
+        }
+
+        /**
+         * Handles the active network.
```

Comment:
This method "doesn't handle active network." It tracks and processes "changes in active network"

Reply:
Changed the comments.

Reply:
The comments are gone, check your commits, you've likely overwritten one commit over another.

Reply:
I can see the changes in the commits.  Probably the changes was from previous commits.